opds_import.py

```diff
@@ -532,27 +540,18 @@ def data_detail_for_feedparser_entry(cls, entry, data_source):
         if not identifier:
             return None, None, None
 
-        failure = None
-        status_code = entry.get('simplified_status_code', None)
-        message = entry.get('simplified_message', None)
-        if status_code is not None and status_code != 200:
-            _db = Session.object_session(data_source)
-            identifier_obj, ignore = Identifier.parse_urn(_db, identifier)
-            error_message = "%s: %s" % (status_code, message)
-            failure = CoverageFailure(identifier_obj, error_message, data_source, transient=True)
-
-        if failure:
-            return identifier, None, failure
-
         # At this point we can assume that we successfully got some
         # metadata, and possibly a link to the actual book.
         try:
             kwargs_meta = cls._data_detail_for_feedparser_entry(entry, data_source)
-            return identifier, kwargs_meta, failure
+            return identifier, kwargs_meta, None
         except Exception, e:
             _db = Session.object_session(data_source)
             identifier_obj, ignore = Identifier.parse_urn(_db, identifier)
-            failure = CoverageFailure(identifier_obj, traceback.format_exc(), data_source, transient=True)
+            failure = CoverageFailure(
+                identifier_obj, traceback.format_exc(), data_source,
+                transient=True
```

Comment:
Now the only time this method can return a CoverageFailure is here, when there's an exception turning the OPDS entry into a Metadata object.